### PR TITLE
Add template and config value for Remote Monitoring compatibility

### PR DESCRIPTION
**Pull request by jillcary**
# Description and Motivation 
In order for Remote Monitoring to update to DS-2.0.1, a new template with the Remote Monitoring default devices needs to be added. Additionally, the config value for userAgent is currently hard-coded to "devicesimulation", this change updates this value to read from an optional environment variable.

This change includes the following:
1. Add a template for Remote Monitoring
1. Update the User Agent string in appsettings.ini to read from an env-var
1. Add default value for the optional User Agent string

# Change type 

- [ ] Bug fix
- [x] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

**Checklist:**

- [x] All tests passed
- [x] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

---
This change is [Reviewable](https://reviewable.io/reviews/azure/device-simulation-dotnet/330)

### CODE REVIEW

File: Services/Devices.cs [hunk 1]
```diff
@@ -98,6 +98,9 @@ public class Devices : IDevices, IDisposable
         // When using bulk operations, this is the max number of devices that the registry APIs allow
         private const int REGISTRY_MAX_BATCH_SIZE = 100;
 
+        // Default string used for user agent if not provided in configuration
+        private const string DEFAULT_USER_AGENT_STRING = "devicesimulation";
```
Comment (dluc) [inline Services/Devices.cs]: pls ask Troy to review this

File: Services/Devices.cs [hunk 2]
```diff
@@ -753,6 +756,11 @@ private IDeviceClientWrapper GetDeviceSdkClient(Device device, IoTHubProtocol pr
         {
             var connectionString = $"HostName={device.IoTHubHostName};DeviceId={device.Id};SharedAccessKey={device.AuthPrimaryKey}";
             var userAgent = this.config.UserAgent;
+            if (string.IsNullOrEmpty(userAgent))
+            {
+                this.log.Debug("Using default string for user agent.", () => new { userAgent });
+                userAgent = DEFAULT_USER_AGENT_STRING;
```
Comment (dluc) [inline Services/Devices.cs]: log also the default

File: Services/Devices.cs [hunk 3]
```diff
@@ -753,6 +756,11 @@ private IDeviceClientWrapper GetDeviceSdkClient(Device device, IoTHubProtocol pr
         {
             var connectionString = $"HostName={device.IoTHubHostName};DeviceId={device.Id};SharedAccessKey={device.AuthPrimaryKey}";
             var userAgent = this.config.UserAgent;
+            if (string.IsNullOrEmpty(userAgent))
+            {
+                this.log.Debug("Using default string for user agent.", () => new { userAgent });
```
Comment (dluc) [inline Services/Devices.cs]: log also the default

File: Services/Devices.cs (hunk 1 above)
Comment (sasathy) [inline Services/Devices.cs]: this can be in appsettings too.

Comment (jillcary) [reply Services/Devices.cs]: Confirmed with @troyhopwood.

File: WebService/Runtime/Config.cs [hunk 4]
```diff
@@ -277,7 +277,7 @@ private static IServicesConfig GetServicesConfig(IConfigData configData)
                 SimulationsStorage = GetStorageConfig(configData, SIMULATIONS_STORAGE_KEY),
                 DevicesStorage = GetStorageConfig(configData, DEVICES_STORAGE_KEY),
                 PartitionsStorage = GetStorageConfig(configData, PARTITIONS_STORAGE_KEY),
-                UserAgent = configData.GetString(USER_AGENT_KEY),
+                UserAgent = configData.GetString(USER_AGENT_KEY, "devicesimulation"),
                 StatisticsStorage = GetStorageConfig(configData, STATISTICS_STORAGE_KEY),
```
Comment (hathind-ms) [inline WebService/Runtime/Config.cs]: please make this a constant and then use here

Comment (jillcary) [reply WebService/Runtime/Config.cs]: Updated.

**Pull request merged**